tests/wallet/test_singleton_store.py

```diff
@@ -0,0 +1,279 @@
+    next_record = SingletonCoinRecord(
+        coin=child_coin,
+        singleton_id=record.singleton_id,
+        wallet_id=record.wallet_id,
+        inner_puzzle=inner_puz,
+        inner_puzzle_hash=inner_puz.get_tree_hash(),
+        confirmed=False,
+        confirmed_at_height=uint32(0),
+        spent_height=uint32(0),
+        lineage_proof=next_lineage_proof,
+        custom_data=record.custom_data,
+        generation=uint32(record.generation + 1),
+        timestamp=uint64(time.time()),
+    )
+    return next_record
+
+
+@pytest.mark.asyncio
+async def test_singleton_store() -> None:
+    async with DBConnection(1) as wrapper:
+        db = await WalletSingletonStore.create(wrapper)
+        record = get_new_singleton_record()
+        singleton_id = record.singleton_id
+        # add record to DB
+        await db.add_singleton_record(record)
+
+        # fetch record by coin_id
+        rec = await db.get_record_by_coin_id(record.coin_id())
+        assert rec == record
+
+        # fetch non-existent coin
+        rec = await db.get_record_by_coin_id(bytes32(token_bytes(32)))
+        assert rec is None
+
+        # make record confirmed
+        confirmed_height = uint32(10)
+        timestamp = uint64(time.time())
+        await db.set_confirmed(record.coin_id(), confirmed_height, timestamp)
+        rec = await db.get_record_by_coin_id(record.coin_id())
+        assert isinstance(rec, SingletonCoinRecord)
+        assert rec.confirmed
+        assert rec.confirmed_at_height == confirmed_height
+
+        # get the next singleton record and add it to DB
+        next_record = get_next_singleton_record(record)
+        await db.add_singleton_record(next_record)
+
+        # check the new record exists and the old record has been set to spent
+        recs = await db.get_records_by_singleton_id(record.singleton_id)
+        assert len(recs) == 2
+        assert not recs[0].confirmed
+        assert recs[1].confirmed
+        assert recs[0].generation == 1
+        assert recs[1].spent_height == 0
+
+        # check get_latest_singleton with and without filtering confirmed
+        latest = await db.get_latest_singleton(singleton_id)
+        assert latest == recs[0]
+        latest = await db.get_latest_singleton(singleton_id, only_confirmed=True)
+        assert latest == recs[1]
+
+        # check get_unconfirmed_singletons
+        unconfirmed = await db.get_unconfirmed_singletons(singleton_id)
+        assert len(unconfirmed) == 1
+        assert unconfirmed[0] == recs[0]
+
+        # confirm the new record
+        await db.set_confirmed(next_record.coin_id(), uint32(20), uint64(time.time()))
+        recs = await db.get_records_by_singleton_id(record.singleton_id)
+        assert recs[0].confirmed
+        assert recs[1].spent_height == recs[0].confirmed_at_height == uint32(20)
+
+        # Delete the last record
+        await db.delete_record_by_coin_id(next_record.coin_id())
```

Comment:
there is no test case that ensures the coin record is gone after calling `delete_record_by_coin_id`.
There's also no test attempting to remove a record that doesn't exist by coin id